### PR TITLE
cp-82: Add user profile sidebar

### PR DESCRIPTION
**Pull request by Ekit18**
image

### CODE REVIEW

File: frontend/src/libs/components/user-profile-sidebar/user-profile-sidebar.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+const UserProfileChatSidebar: React.FC = () => {
+  const { authenticatedUser } = useAppSelector(({ auth }) => ({
+    authenticatedUser: auth.authenticatedUser,
+  }));
+
+  return (
+    <div className={styles['container']}>
+      <div className={styles['header']}>Profile</div>
+      <div className={styles['user-info']}>
+        <img src="images/avatar.png" alt="not found" />
```
Comment (fshabanov) [inline frontend/src/libs/components/user-profile-sidebar/user-profile-sidebar.tsx]:
```suggestion
        <img src="images/avatar.png" alt="avatar" />
```

File: frontend/public/images/avatar.png
Comment (fshabanov) [inline frontend/public/images/avatar.png:1]: Let's use different default avatar. Something similar to this.
image

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: shared/src/packages/users/libs/types/user-auth-response-dto.type.ts
Comment (Xxandder) [inline shared/src/packages/users/libs/types/user-auth-response-dto.type.ts:1]: Why did you add here createdAt and updatedAt fields?

**Review by Xxandder — CHANGES_REQUESTED** [review]
Everithing is good)

File: shared/src/packages/users/libs/types/user-auth-response-dto.type.ts
Comment (Ekit18) [reply shared/src/packages/users/libs/types/user-auth-response-dto.type.ts:1]: it comes from backend
image

Comment (Xxandder) [reply shared/src/packages/users/libs/types/user-auth-response-dto.type.ts:1]: For me it works without adding these fields in userAuthResoinseDto, It's kinda strange)

Comment (Xxandder) [reply shared/src/packages/users/libs/types/user-auth-response-dto.type.ts:1]: Good notice, bacause It worked but was not good to have userRequestDto without these fields

File: frontend/public/images/avatar.png
Comment (Ekit18) [reply frontend/public/images/avatar.png:1]: done

File: frontend/src/libs/components/user-profile-sidebar/user-profile-sidebar.tsx (hunk 1 above)
Comment (Ekit18) [reply frontend/src/libs/components/user-profile-sidebar/user-profile-sidebar.tsx]: done

File: frontend/public/images/avatar.png
Comment (fshabanov) [reply frontend/public/images/avatar.png:1]: I see that you have added exactly this image. It does not have a transparent background, have you checked how it looks?

**Review by iamAlinaaa — DISMISSED** [review]

File: frontend/src/libs/components/user-profile-sidebar/user-profile-sidebar.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { useAppSelector } from '#libs/hooks/hooks.js';
```
Comment (liza-veis) [inline frontend/src/libs/components/user-profile-sidebar/user-profile-sidebar.tsx]: move it to `pages/user-profile/components/profile-sidebar`

**Review by liza-veis — CHANGES_REQUESTED** [review]

File: frontend/src/libs/components/user-profile-sidebar/user-profile-sidebar.tsx (hunk 2 above)
Comment (Ekit18) [reply frontend/src/libs/components/user-profile-sidebar/user-profile-sidebar.tsx]: done

**Comment by juanc-escobar** [comment]
the implementation looks good and well organized ✅

**Review by OlenaYatseniuk — DISMISSED** [review]
Looks good to me!

File: frontend/src/pages/user-profile/user-profile.tsx [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { UserProfileChatSidebar } from './components/components.js';
+
+const UserProfile: React.FC = () => {
+  return (
+    <>
+      <UserProfileChatSidebar />
+    </>
+  );
```
Comment (HIMDanny) [inline frontend/src/pages/user-profile/user-profile.tsx]: Maybe let's do it this way?
```suggestion
  return <UserProfileChatSidebar />;
```

Comment (Ekit18) [reply frontend/src/pages/user-profile/user-profile.tsx]: it is one of the component on this page. so there will be in the future like div element istead of <></>

Comment (HIMDanny) [reply frontend/src/pages/user-profile/user-profile.tsx]: If it will be changed anyway, I think it is better for now to keep it clear

File: frontend/public/images/avatar.png
Comment (Ekit18) [reply frontend/public/images/avatar.png:1]: done
image

**Review by dmytro-kucherenko — DISMISSED** [review]

**Review by HIMDanny — DISMISSED** [review]

File: frontend/src/pages/user-profile/components/user-profile-sidebar/user-profile-sidebar.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import { useAppSelector } from '#libs/hooks/hooks.js';
+
+import styles from './styles.module.scss';
+
+const UserProfileChatSidebar: React.FC = () => {
```
Comment (liza-veis) [inline frontend/src/pages/user-profile/components/user-profile-sidebar/user-profile-sidebar.tsx]:
```suggestion
const UserProfileSidebar: React.FC = () => {
```

File: frontend/src/pages/user-profile/components/user-profile-sidebar/styles.module.scss [hunk 5]
```diff
@@ -0,0 +1,46 @@
+.container {
+  width: 350px;
+  height: 100vh;
```
Comment (liza-veis) [inline frontend/src/pages/user-profile/components/user-profile-sidebar/styles.module.scss]: remove all the vh in this file, it's better to avoid them and use only when they needed

**Review by liza-veis — CHANGES_REQUESTED** [review]

File: frontend/src/pages/user-profile/components/user-profile-sidebar/styles.module.scss (hunk 5 above)
Comment (Ekit18) [reply frontend/src/pages/user-profile/components/user-profile-sidebar/styles.module.scss]: done

File: frontend/src/pages/user-profile/components/user-profile-sidebar/user-profile-sidebar.tsx (hunk 4 above)
Comment (Ekit18) [reply frontend/src/pages/user-profile/components/user-profile-sidebar/user-profile-sidebar.tsx]: done

**Review by Xxandder — DISMISSED** [review]

**Review by iamAlinaaa — DISMISSED** [review]

**Review by dmytro-kucherenko — DISMISSED** [review]

**Review by OlenaYatseniuk — DISMISSED** [review]

**Review by OlesiaTr — DISMISSED** [review]

**Review by HIMDanny — DISMISSED** [review]

**Review by fshabanov — DISMISSED** [review]

**Review by Nanaier — DISMISSED** [review]

**Review by dmytro-kucherenko — DISMISSED** [review]

**Review by dmytro-kucherenko — DISMISSED** [review]

File: frontend/src/pages/user-profile/components/user-profile-sidebar/user-profile-sidebar.tsx [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+      <div className={styles['user-info']}>
+        <div className={styles['avatar-container']}>
+          <img
+            src="images/avatar.png"
+            alt="avatar"
+            className={styles['avatar']}
+          />
+        </div>
+        <span className={styles['user-name']}>
+          {authenticatedUser?.fullName}
```
Comment (fshabanov) [inline frontend/src/pages/user-profile/components/user-profile-sidebar/user-profile-sidebar.tsx]: We will always have a user here, cast the type

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: frontend/src/pages/user-profile/components/user-profile-sidebar/user-profile-sidebar.tsx (hunk 6 above)
Comment (Ekit18) [reply frontend/src/pages/user-profile/components/user-profile-sidebar/user-profile-sidebar.tsx]: done

**Review by BrianSammit — DISMISSED** [review]

**Review by dmytro-kucherenko — DISMISSED** [review]

**Review by Vika030718 — DISMISSED** [review]

File: frontend/src/libs/enums/app-route.enum.ts [hunk 7]
```diff
@@ -2,6 +2,7 @@ const AppRoute = {
   ROOT: '/',
   SIGN_IN: '/sign-in',
   SIGN_UP: '/sign-up',
+  USER_PROFILE: '/user-profile',
```
Comment (fshabanov) [inline frontend/src/libs/enums/app-route.enum.ts]:
```suggestion
  PROFILE: '/profile',
```

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: frontend/src/libs/enums/app-route.enum.ts (hunk 7 above)
Comment (Ekit18) [reply frontend/src/libs/enums/app-route.enum.ts]: done

File: frontend/src/pages/user-profile/components/user-profile-sidebar/user-profile-sidebar.tsx [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+    authenticatedUser: auth.authenticatedUser,
+  }));
+
+  return (
+    <div className={styles['container']}>
+      <div className={styles['header']}>Profile</div>
+      <div className={styles['user-info']}>
+        <div className={styles['avatar-container']}>
+          <img
+            src="images/avatar.png"
```
Comment (v0ldemar01) [inline frontend/src/pages/user-profile/components/user-profile-sidebar/user-profile-sidebar.tsx]: First import that component and reuse it here

**Review by v0ldemar01 — CHANGES_REQUESTED** [review]

File: frontend/src/pages/user-profile/components/user-profile-sidebar/user-profile-sidebar.tsx (hunk 8 above)
Comment (Ekit18) [reply frontend/src/pages/user-profile/components/user-profile-sidebar/user-profile-sidebar.tsx]: done

File: frontend/src/assets/img/profile.svg
Comment (liza-veis) [inline frontend/src/assets/img/profile.svg:1]: move to icons and replace color with currentColor

File: frontend/src/assets/img/avatar.png
Comment (liza-veis) [inline frontend/src/assets/img/avatar.png:1]: rename to avatar-placeholder

File: frontend/src/index.tsx [hunk 9]
```diff
@@ -44,6 +45,16 @@ createRoot(document.querySelector('#root') as HTMLElement).render(
...
               },
+              {
+                path: AppRoute.PROFILE,
+                element: (
+                  <ProtectedRoute>
+                    <SidebarWrapper>
+                      <UserProfile />
+                    </SidebarWrapper>
+                  </ProtectedRoute>
+                ),
```
Comment (liza-veis) [inline frontend/src/index.tsx]: Let's rename it to dashboard, also it should be our root component, so put it at the top instead of root
```suggestion
                path: AppRoute.ROOT,
                element: (
                  <ProtectedRoute>
                    <SidebarWrapper>
                      <Dashboard />
                    </SidebarWrapper>
                  </ProtectedRoute>
                ),
```

File: frontend/src/libs/components/sidebar-wrapper/libs/constants.ts [hunk 10]
```diff
@@ -1,9 +1,11 @@
 import home from '#assets/img/home.svg';
+import profile from '#assets/img/profile.svg';
 import { AppRoute } from '#libs/enums/enums.js';
 import { type Route } from '#libs/types/types.js';
 
 const SIDEBAR_ROUTES: Route[] = [
   { path: AppRoute.ROOT, name: 'home', icon: home },
+  { path: AppRoute.PROFILE, name: 'profile', icon: profile },
```
Comment (liza-veis) [inline frontend/src/libs/components/sidebar-wrapper/libs/constants.ts]:
```suggestion
  { path: AppRoute.ROOT, name: 'Home', icon: home },
```

File: frontend/src/pages/user-profile/components/user-profile-sidebar/styles.module.scss [hunk 11]
```diff
@@ -0,0 +1,46 @@
+.container {
+  width: 350px;
```
Comment (liza-veis) [inline frontend/src/pages/user-profile/components/user-profile-sidebar/styles.module.scss]: don't set fixed width, we should make everything responsive

File: frontend/src/pages/user-profile/components/user-profile-sidebar/styles.module.scss [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+  height: 10%;
+  padding: 0 20px;
+  font-weight: 600;
+  font-size: 20px;
+  border-bottom: 1px solid var(--blue-100);
+}
+
+.avatar-container {
+  width: 72px;
+  height: 72px;
```
Comment (liza-veis) [inline frontend/src/pages/user-profile/components/user-profile-sidebar/styles.module.scss]: same about width and height, if it's related to image then add it via attributes

File: frontend/src/pages/user-profile/components/user-profile-sidebar/styles.module.scss [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+
+.avatar {
+  width: 100%;
+}
+
+.user-info {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  height: 90%;
```
Comment (liza-veis) [inline frontend/src/pages/user-profile/components/user-profile-sidebar/styles.module.scss]:
```suggestion
```

File: frontend/src/pages/user-profile/components/user-profile-sidebar/user-profile-sidebar.tsx [hunk 14]
```diff
@@ -0,0 +1,29 @@
+import avatar from '#assets/img/avatar.png';
+import { useAppSelector } from '#libs/hooks/hooks.js';
+import { type UserAuthResponseDto } from '#packages/users/users.js';
+
+import styles from './styles.module.scss';
+
+const UserProfileSidebar: React.FC = () => {
```
Comment (liza-veis) [inline frontend/src/pages/user-profile/components/user-profile-sidebar/user-profile-sidebar.tsx]: Let's rename it to UserProfileInfo
```suggestion
const UserProfileInfo: React.FC = () => {
```

**Review by liza-veis — CHANGES_REQUESTED** [review]

File: frontend/src/pages/user-profile/components/user-profile-sidebar/user-profile-sidebar.tsx (hunk 14 above)
Comment (Ekit18) [reply frontend/src/pages/user-profile/components/user-profile-sidebar/user-profile-sidebar.tsx]: done

File: frontend/src/pages/user-profile/components/user-profile-sidebar/styles.module.scss (hunk 13 above)
Comment (Ekit18) [reply frontend/src/pages/user-profile/components/user-profile-sidebar/styles.module.scss]: done

File: frontend/src/pages/user-profile/components/user-profile-sidebar/styles.module.scss (hunk 12 above)
Comment (Ekit18) [reply frontend/src/pages/user-profile/components/user-profile-sidebar/styles.module.scss]: done

File: frontend/src/pages/user-profile/components/user-profile-sidebar/styles.module.scss (hunk 11 above)
Comment (Ekit18) [reply frontend/src/pages/user-profile/components/user-profile-sidebar/styles.module.scss]: done

File: frontend/src/libs/components/sidebar-wrapper/libs/constants.ts (hunk 10 above)
Comment (Ekit18) [reply frontend/src/libs/components/sidebar-wrapper/libs/constants.ts]: done

File: frontend/src/index.tsx (hunk 9 above)
Comment (Ekit18) [reply frontend/src/index.tsx]: done

File: frontend/src/assets/img/avatar.png
Comment (Ekit18) [reply frontend/src/assets/img/avatar.png:1]: done

File: frontend/src/assets/img/profile.svg
Comment (Ekit18) [reply frontend/src/assets/img/profile.svg:1]: done

File: frontend/src/libs/enums/app-route.enum.ts [hunk 15]
```diff
@@ -2,6 +2,7 @@ const AppRoute = {
   ROOT: '/',
   SIGN_IN: '/sign-in',
   SIGN_UP: '/sign-up',
+  PROFILE: '/profile',
```
Comment (liza-veis) [inline frontend/src/libs/enums/app-route.enum.ts]:
```suggestion
```

**Review by liza-veis — CHANGES_REQUESTED** [review]

File: frontend/src/libs/enums/app-route.enum.ts (hunk 15 above)
Comment (Ekit18) [reply frontend/src/libs/enums/app-route.enum.ts]: done

**Review by liza-veis — DISMISSED** [review]

**Review by fshabanov — DISMISSED** [review]

**Review by liza-veis — DISMISSED** [review]

**Review by v0ldemar01 — CHANGES_REQUESTED** [review]
You should wait for successful finishing ci before re-requesting the review

File: frontend/src/pages/dashboard/components/user-profile-info/styles.module.scss [hunk 16]
```diff
@@ -0,0 +1,45 @@
...
+    var(--blue-100) 100%
+  );
+}
+
+.header {
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  align-items: center;
+  height: 10%;
```
Comment (Xxandder) [inline frontend/src/pages/dashboard/components/user-profile-info/styles.module.scss]: We should avoid relative values for layout

**Review by Xxandder — CHANGES_REQUESTED** [review]

File: frontend/src/pages/dashboard/components/user-profile-info/styles.module.scss (hunk 16 above)
Comment (Ekit18) [reply frontend/src/pages/dashboard/components/user-profile-info/styles.module.scss]: @Liza-Veis so should i use px instead of percentage?

Comment (liza-veis) [reply frontend/src/pages/dashboard/components/user-profile-info/styles.module.scss]: which element is it?

Comment (liza-veis) [reply frontend/src/pages/dashboard/components/user-profile-info/styles.module.scss]: for header we don't need this at all, it's better to achieve it using padding

Comment (Ekit18) [reply frontend/src/pages/dashboard/components/user-profile-info/styles.module.scss]: done

**Review by OlesiaTr — APPROVED** [review]

**Review by liza-veis — APPROVED** [review]

**Review by BrianSammit — APPROVED** [review]

**Review by fshabanov — APPROVED** [review]
